### PR TITLE
[SYCL][FPGA][NFC] Refactor [[intel::num_simd_work_items()]] attribute tests

### PR DESCRIPTION
**Pull request by smanna12**
This patch updates tests for [[intel::num_simd_work_items()]] attribute:

1. separates AST and diagnostics to make it easier to read/follow-up.
2. removes duplicate test cases.
3. adds missing tests.
4. no compiler changes.

Signed-off-by: Soumi Manna <soumi.manna@intel.com>

### CODE REVIEW

**Comment by smanna12** [comment]
The failures are not related to my PR.

**Comment by smanna12** [comment]
ping @premanandrao / @elizabethandrews

File: clang/test/SemaSYCL/num_simd_work_items.cpp [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+  void
+  operator()() const {}
+};
+
+// FIXME: This should be accepted instead of error which turns out to be
+// an implementation bug that shouldn't be visible to the user as there
+// aren't really any default values. The dimensionality of the attribute
+// must match the kernel, so three different forms of the attribute
+// (one, two, and three argument) can be used instead of assuming default
+// values. This will prevent to redeclare the function with a different dimensionality.
```
Comment (premanandrao) [inline clang/test/SemaSYCL/num_simd_work_items.cpp:72]:
```suggestion
// values. This prevents redeclaring the function with a different dimensionality.
```

File: clang/test/SemaSYCL/num_simd_work_items.cpp [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
```
Comment (premanandrao) [inline clang/test/SemaSYCL/num_simd_work_items.cpp:85]:
```suggestion
// values. This prevents redeclaring the function with a different dimensionality.
```

File: clang/test/SemaSYCL/num_simd_work_items.cpp [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+
+    h.single_task<class test_kernel21>(
+        []() [[intel::num_simd_work_items(1), intel::num_simd_work_items(2)]] {}); // expected-warning{{attribute 'num_simd_work_items' is already applied with different arguments}}  // expected-note {{previous attribute is here}}
+
+    h.single_task<class test_kernel22>(TRIFuncObjBad19());
+  });
+  return 0;
+}
+
+// Test that checkes template parameter support for 'num_simd_work_items' attribute on sycl device.
```
Comment (premanandrao) [inline clang/test/SemaSYCL/num_simd_work_items.cpp:264]:
```suggestion
// Test that checks template parameter support for 'num_simd_work_items' attribute on sycl device.
```

**Review by premanandrao — COMMENTED** [review]
Minor nits.

File: clang/test/SemaSYCL/num_simd_work_items.cpp (hunk 3 above)
Comment (smanna12) [reply clang/test/SemaSYCL/num_simd_work_items.cpp]: removed. Not needed.

**Comment by smanna12** [comment]
Thanks @premanandrao for reviews.

https://github.com/orgs/intel/teams/llvm-gatekeepers, this PR is ready for a merge. Failures are unrelated to this PR. Thank you

**Comment by smanna12** [comment]
@intel/llvm-gatekeepers, this PR is ready for a merge. Failures are unrelated to the PR.

**Pull request merged**